src/login/inhibit.c

```diff
@@ -277,6 +277,9 @@ int main(int argc, char *argv[]) {
                         _exit(EXIT_FAILURE);
                 }
 
+                /* Ignore SIGINT and allow the forked process to receive it */
+                (void) ignore_signals(SIGINT, -1);
```

Comment:
makes sense, but I think we should move this further up, to the beginning of the big "else" branch, so that the parent process has the signal set to ignore during the entire time the child exists, from its earliest instant one to the very end. This closes a (mostly theoretical) race that when the child process tells the user it's doing something, and the user hits C-c that we haven't yet masked SIGINT in the parent.

Note that the signal is automatically reset in the child, given that we pass `FORK_RESET_SIGNALS` to `safe_fork()`.